### PR TITLE
Feature/newtypes

### PR DESCRIPTION
**Pull request by noambergIL**
new types, no panic in marshal 
also some changes by itamar to make this repo gomod .

### CODE REVIEW

File: .circleci/main.sh [hunk 1]
```diff
@@ -1,33 +1,23 @@
 #!/bin/bash -xe
 
 PROJ_PATH=`pwd`
+GO_VERSION="1.12.6"
 
 # First let's install Go 1.11
-echo "Installing Go 1.11"
+echo "Installing Go 1.12..."
```
Comment (netoneko) [inline .circleci/main.sh:7]: Just put $GO_VERSION here as well

File: .circleci/main.sh [hunk 2]
```diff
@@ -1,33 +1,23 @@
...
+curl -O https://dl.google.com/go/go${GO_VERSION}.linux-amd64.tar.gz
+sudo tar -xf go${GO_VERSION}.linux-amd64.tar.gz
+
 # Uninstall older version of Go
 sudo rm -rf /usr/local/go
 sudo mv go /usr/local
 
 export GOROOT=/usr/local/go
-export GOPATH=$PROJ_PATH
+#export GOPATH=$PROJ_PATH
```
Comment (netoneko) [inline .circleci/main.sh:18]: If you're not using it, remove it

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,11 @@
+module github.com/orbs-network/gamma-cli
+
+go 1.13
+
+require (
+	github.com/orbs-network/lean-helix-go v0.2.3
```
Comment (netoneko) [inline go.mod:6]: Why lean helix?

File: test/restart_test.go [hunk 4]
```diff
@@ -58,18 +58,3 @@ func TestStopAfterCrashOfGammaServer(t *testing.T) {
 	require.NoError(t, err, "stop Gamma server should succeed")
 	require.True(t, strings.Contains(out, "Prism blockchain explorer stopped."), "Prism server should stop even if gamma crashed")
 }
-
-func TestStartedButNotReadyMessage(t *testing.T) {
```
Comment (netoneko) [inline test/restart_test.go:62]: Why did you remove the test?

**Review by netoneko — CHANGES_REQUESTED** [review]

File: test/restart_test.go (hunk 4 above)
Comment (noambergIL) [reply test/restart_test.go:62]: This test is flaky because it runs gamma-server and hopes to call it before it runs. It seems too specific.

File: go.mod (hunk 3 above)
Comment (noambergIL) [reply go.mod:6]: it is need for block proof in crypto/reciept_proof.go
